extensions/src/main/java/org/pentaho/platform/web/servlet/UploadFileUtils.java

```diff
@@ -586,6 +588,32 @@ private long getFolderSize( File folder ) {
     return foldersize;
   }
 
+  /**
+   * <p>Removes the file name, leaving all extensions of a given filename.</p>
+   * <p>Returns <code>null</code> if filename is <code>null</code>, and an empty string if the filename is empty or has
+   * no extensions.</p>
+   *
+   * @param filename the filename for which we want the extensions
+   * @return all extensions of the given filename
+   */
+  protected String removeFileName( String filename ) {
+    if ( null != filename ) {
+      int dotIndex = filename.indexOf( DOT );
+
+      if ( 0 < dotIndex ) {
+        int len = filename.length();
+        ++dotIndex;
+        if ( len > dotIndex ) {
+          return filename.substring( dotIndex );
```

Comment:
I'd like to believe that `commons-io` would have some method to extract the extension of a file.

Reply:
Yes and no, technically extension refers to the last block of text after the last dot (and for that there's the `getExtension` method of `common-io`.
There's, however, no method that returns "all" extensions as explained in the comment above.